### ISSUE TITLE
Fulcro defsc unused namespace warning

### ISSUE THREAD

**Issue by codonnell** [borkdude/clj-kondo#858]
**version** v2020.04.05

**platform** linux using native

**problem**

An unused namespace warning is triggered when that namespace is used in a component options map in fulcro's `defsc` form. In addition, despite linting `defsc` the same as `defn`, the forms result in different lint results when interchanged.

**repro**

```clojure
(ns my-app
  (:require [com.fulcrologic.fulcro.data-fetch :as df]))

;; Unused namespace: df
(defsc Home [_this _]
  {:query []
   :ident nil
   :initial-state nil
   :route-segment []
   :will-enter (fn [app _]
                 (df/load! app [:component/id :created-gift-lists]
                   ui.gift-list/CreatedGiftLists))}
  (dom/div {} "Home"))
```

```clojure
(ns my-app
  (:require [com.fulcrologic.fulcro.data-fetch :as df]))

;; No warning
(defsc Home [_this _]
  {:lifecycle-method (df/load! "foo")}
  (dom/div {} "Home"))

;; Unused namespace: df
(defn Home [_this _]
  {:lifecycle-method (df/load! "foo")}
  (dom/div {} "Home"))
```

**config**

```clojure
{:lint-as {com.fulcrologic.fulcro.components/defsc clojure.core/defn}}
```

**expected behavior**

I would expect not to get any unused namespace warnings in the scenarios above.

**Comment by borkdude** [comment]
@codonnell I'm using this code in a file called fulcro.clj:

```
(ns my-app
  {:clj-kondo/config '{:lint-as {fulcro/defsc clojure.core/defn}}}
  (:require [df :as df]
            [fulcro :as fulcro]
            [dom :as dom]))

(fulcro/defsc Home1 [_this _]
  {:query []
   :ident nil
   :initial-state nil
   :route-segment []
   :will-enter (fn [app _]
                 (df/load! app [:component/id :created-gift-lists]
                           ui.gift-list/CreatedGiftLists))}
  (dom/div {} "Home"))

;; No warning
(fulcro/defsc Home2 [_this _]
  {:lifecycle-method (df/load! "foo")}
  (dom/div {} "Home"))

;; Unused namespace: df
(defn Home3 [_this _]
  {:lifecycle-method (df/load! "foo")}
  (dom/div {} "Home"))
```

I see no warnings:

```
$ /usr/local/bin/clj-kondo --lint /tmp/fulcro.clj
linting took 13ms, errors: 0, warnings: 0
```

Am I missing something?

**Comment by codonnell** [comment]
Sorry, should have been more clear in my report. First of all, I can no longer reproduce the `Home3` case; perhaps this was fixed as part of the 05.02 release?

In order to reproduce the `Home1` problem, I first comment out (or delete, since there is no longer a problem) the `Home2` and `Home3` forms. These were meant to be evaluated as separate lint cases. Now with the file `/tmp/kondo.clj`:

```clojure
(ns kondo
  (:require [df :as df]
            [fulcro :as fulcro]
            [ui.gift-list :as ui.gift-list]
            [dom :as dom]))

(fulcro/defsc Home1 [_this _]
  {:query []
   :ident nil
   :initial-state nil
   :route-segment []
   :will-enter (fn [app _]
                 (df/load! app [:component/id :created-gift-lists]
                   ui.gift-list/CreatedGiftLists))}
  (dom/div {} "Home"))
```

and `/tmp/.clj-kondo/config.edn` as:
```clojure
{:lint-as {fulcro/defsc clojure.core/defn}}
```

I run `clj-kondo --lint kondo.clj` and get the errors:
```
kondo.clj:2:14: warning: namespace df is required but never used
kondo.clj:4:14: warning: namespace ui.gift-list is required but never used
```

Interestingly, if I inline the config (which I didn't realize was possible!) I cannot reproduce the problem.

**Comment by borkdude** [comment]
Can reproduce it now. Thanks.

**Comment by borkdude** [comment]
@codonnell This time I think I got it. Can you test with the latest commit on master?

**Comment by codonnell** [comment]
Fixed on master for me (via `clj -Sdeps '{:deps {clj-kondo {:git/url "https://github.com/borkdude/clj-kondo.git" :sha "6e0b64fe6977017b954717c83995fb2c7687cf13"}}}' -m clj-kondo.main --lint kondo.clj`).

**Comment by codonnell** [comment]
Thanks!

**Comment by borkdude** [comment]
@codonnell Until the next release, you can use this linux binary from CircleCI master:

https://9499-176829714-gh.circle-artifacts.com/0/release/clj-kondo-2020.05.03-SNAPSHOT-linux-amd64.zip
